### PR TITLE
[webview_flutter] Setup XCTests

### PR DESCRIPTION
**Pull request by cyanglaz**
## Description

We need XCTests in the webview_flutter plugin to test certain things that dart side integration tests cannot. This PR sets up some basic tests.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/webview_flutter/ios/Tests/FLTWebViewTests.m [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+@interface FLTWebViewTests : XCTestCase
+
+@property(strong, nonatomic) NSObject<FlutterBinaryMessenger> *mockBinaryMessenger;
+
+@end
+
+@implementation FLTWebViewTests
+
+- (void)setUp {
+  self.mockBinaryMessenger = OCMProtocolMock(@protocol(FlutterBinaryMessenger));
```
Comment (jmagman) [inline packages/webview_flutter/ios/Tests/FLTWebViewTests.m]:
```
[super setUp];
```

File: packages/webview_flutter/ios/Tests/FLTWebViewTests.m [hunk 2]
```diff
@@ -0,0 +1,38 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import <Flutter/Flutter.h>
+#import <OCMock/OCMock.h>
+#import <XCTest/XCTest.h>
```
Comment (jmagman) [inline packages/webview_flutter/ios/Tests/FLTWebViewTests.m]:
```
@import Flutter;
@import OCMock;
@import XCTest;
```

File: packages/webview_flutter/example/ios/Runner.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -316,6 +394,53 @@
...
+			isa = XCBuildConfiguration;
+			buildSettings = {
+				BUNDLE_LOADER = "$(TEST_HOST)";
+				CLANG_ANALYZER_NUMBER_OBJECT_CONVERSION = YES_AGGRESSIVE;
+				CLANG_CXX_LANGUAGE_STANDARD = "gnu++14";
+				CLANG_ENABLE_OBJC_WEAK = YES;
+				CLANG_WARN_DOCUMENTATION_COMMENTS = YES;
+				CLANG_WARN_UNGUARDED_AVAILABILITY = YES_AGGRESSIVE;
+				CODE_SIGN_STYLE = Automatic;
+				DEVELOPMENT_TEAM = S8QB4VV633;
```
Comment (jmagman) [inline packages/webview_flutter/example/ios/Runner.xcodeproj/project.pbxproj]: Don't commit the DEVELOPMENT_TEAM to the repo.

File: packages/webview_flutter/example/ios/Runner.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -176,9 +226,16 @@
 		97C146E61CF9000F007C117D /* Project object */ = {
 			isa = PBXProject;
 			attributes = {
+				DefaultBuildSystemTypeForWorkspace = Original;
 				LastUpgradeCheck = 1030;
 				ORGANIZATIONNAME = "The Chromium Authors";
 				TargetAttributes = {
+					68BDCAE823C3F7CB00D9C032 = {
+						CreatedOnToolsVersion = 11.3;
+						DevelopmentTeam = S8QB4VV633;
```
Comment (jmagman) [inline packages/webview_flutter/example/ios/Runner.xcodeproj/project.pbxproj]: Don't commit the DEVELOPMENT_TEAM to the repo.

**Comment by KevinTheGray** [comment]
@cyanglaz I'm trying to add some tests for the video_player plugin, and I'm wondering, how do you run these tests locally?  I'm trying `pod lib lint webview_flutter.podspec --verbose`  It's failing with an error 
`rdqgnr-webview_flutter/Pods/Target Support Files/webview_flutter/webview_flutter-Unit-Tests-frameworks.sh: line 141: ARCHS[@]: unbound variable`

Any ideas what to do there?

**Comment by jmagman** [comment]
> @cyanglaz I'm trying to add some tests for the video_player plugin, and I'm wondering, how do you run these tests locally? I'm trying `pod lib lint webview_flutter.podspec --verbose` It's failing with an error
> `rdqgnr-webview_flutter/Pods/Target Support Files/webview_flutter/webview_flutter-Unit-Tests-frameworks.sh: line 141: ARCHS[@]: unbound variable`
> 
> Any ideas what to do there?

Run with `--no-clean` so the Xcode project gets preserved, then run the tests in Xcode instead to see what's going on.  Some valid ARCH not getting set?

**Comment by KevinTheGray** [comment]
@cyanglaz I tried with `--no-clean` and then running in XCode gives me `Module 'OCMock' not found` for `FLTWebViewTests.m`

**Comment by KevinTheGray** [comment]
what did you use to run these tests locally?  Is there a certain directory I should be running this from?  Sorry, not very familiar with running XCode tests in this manner.  

It seems like the the example project would need knowledge of the OCMock dependency, right?

**Pull request merged**